models/users.js

```diff
@@ -540,6 +545,30 @@ const getUsersBasedOnFilter = async (query) => {
   return [];
 };
 
+const getUsersWithOnboardingState = async (query) => {
+  const time = query.time;
+  const range = Number(time.split("d")[0]);
+  const { allUserStatus } = await getAllUserStatus(query);
+  const allUsersWithOnboardingState = filterUsersWithOnboardingState(allUserStatus);
```

Comment:
instead of fetching all user state and then filtering out Onboarding. cant we nor directly get onboarding state from DB ?